Add CLI options table (#11187)
trinodb/trino: docs/src/main/sphinx/installation/cli.rst

The CLI requires a Java virtual machine available on the path.
It can be used with Java version 8 and higher.

The CLI uses the HTTP protocol and the :doc:`Trino client REST API
</develop/client-protocol>` to communicate with the coordinator on the Trino
cluster.

Installation
------------

Download :maven_download:`cli`, rename it to ``trino``, make it executable with
``chmod +x``, and run it to show the version of the CLI:

.. code-block:: text

    ./trino --version

Run the CLI with ``--help`` or ``-h`` to see all available options.

Running the CLI
---------------

The minimal command to start the CLI in interactive mode specifies the URL of
the coordinator in the Trino cluster:

.. code-block:: text

    ./trino --server http://trino.example.com:8080

> Many other options are available to further configure how to start the CLI:

Reviewer: ```suggestion
Many other options are available to further configure the CLI which can be listed by running `./trino --help`
```

.. list-table::
  :widths: 40, 60

>   :header-rows: 1

Reviewer: Remove this table.

Reviewer 2: this table assembles all the options that are useful for starting the CLI that are generic .. we want to build up the section above with more details and examples for it all and therefore need this table. We also want to expand the content in the table.

And then in addition we want to add a section about batch mode with examples and a table for all the related parameters for that .. we just want to confirm that this is okay before we build all that content

Reviewer: Adding details about configs which isn't given by --help is okay. That is not what this table is though right now. It's just a single large table with mostly non-essential options (all of which can already be seen by running trino --help).

Reviewer 2: Yeah .. we want to build this out more .. I will work with @Jessie212

  * - Option
    - Description
  * - ``--catalog``
    - Sets the default catalog. You can change the default catalog and schema
      with :doc:`/sql/use`.
  * - ``--client-info``
    - Add extra information about the client.
  * - ``--client-request-timeout``
    - Client request timeout (default: 2m).
  * - ``--client-tags``
    - Add extra tags information about the client, and the CLI user. The tags
      can be used as input for resource groups. :doc:`/admin/resource-groups`.
  * - ``--debug``
    - Enable debug information for :ref:`cli-troubleshooting`.
  * - ``--disable-compression``
    - Disable compression of query results.
  * - ``--editing-mode``

>     - Allows switching key bindings in the CLI.

Reviewer 2: you need to add vi or emacs as valid choices here and also say what the default is

You can now run SQL statements. After processing, the CLI will show results and
statistics.

.. code-block:: text

  trino> SELECT count(*) FROM tpch.tiny.nation;

  _col0
  -------
      25
  (1 row)

  Query 20220324_213359_00007_w6hbk, FINISHED, 1 node
  Splits: 13 total, 13 done (100.00%)
  2.92 [25 rows, 0B] [8 rows/s, 0B/s]

As part of starting the CLI, you can set the default catalog and schema. This
allows you to query tables directly without specifying catalog and schema.

.. code-block:: text

  ./trino --server http://trino.example.com:8080 --catalog tpch --schema tiny

  trino:tiny> SHOW TABLES;

    Table
  ----------
  customer
  lineitem
  nation
  orders
  part
  partsupp
  region
  supplier
  (8 rows)

You can also set the default catalog and schema with the :doc:`/sql/use`
statement.

.. code-block:: text

  trino> USE tpch.tiny;
  USE
  trino:tiny>

Many other options are available to further configure the CLI in interactive
mode:

.. list-table::
  :widths: 40, 60
  :header-rows: 1

  * - Option
    - Description
  * - ``--catalog``
    - Sets the default catalog. You can change the default catalog and schema
      with :doc:`/sql/use`.
  * - ``--client-info``

>     - Adds extra information about the client.

Reviewer: Maybe mention this accepts arbitrary text values?

  * - ``--client-request-timeout``
    - Sets the duration for query processing, after which, the client request is
      terminated. Defaults to ``2m``.
  * - ``--client-tags``

>     - Adds extra tags information about the client and the CLI user. The tags

Reviewer: Maybe mention this accepts comma-separated values.

      with the password value to avoid the prompt. For more information, see :ref:`cli-username-password-auth`.
  * - ``--schema``
    - Sets the default schema. You can change the default catalog and schema
      with :doc:`/sql/use`.
  * - ``--server``
    - The HTTP/HTTPS address and port of the Trino coordinator. The port must be
      set to the port the Trino coordinator is listening for connections on.
      Trino server location defaults to ``http://localhost:8080``.
  * - ``--session``
    - Sets one or more :ref:`session properties
      <session-properties-definition>`. Property can be used multiple times with
      the format ``session_property_name=value``.
  * - ``--socks-proxy``
    - Configures the URL of the SOCKS proxy to connect to Trino.
  * - ``--source``
    - Specifies the name of the application or source connecting to Trino.
      Defaults to ``trino-cli``. The value can be used as input for
      :doc:`/admin/resource-groups`.
  * - ``--timezone``
    - Sets the time zone for the session using the `time zone name
      <https://en.wikipedia.org/wiki/List_of_tz_database_time_zones>`_. Defaults

>       to ``America/New_York``.

Reviewer: Default is wrong. It defaults to JVM timezone (which happens to be America/New_York for people running JVMs in that timezone).

>       to ``America/New_York``.

Reviewer 2: Updated a bit

The CLI requires a Java virtual machine available on the path.
It can be used with Java version 8 and higher.

> The CLI uses the HTTP protocol and the :doc:`Trino client REST API

Reviewer: ```suggestion
The CLI uses the :doc:`Trino client REST API
```

HTTP is implementation detail of the Trino REST API, may not always be so. If we want to keep it maybe word as `The CLI uses the Trino client REST API over HTTP/S`?

Reviewer 2: lets reword as suggested with HTTP/HTTPS 

the reason we want to have this is that it lets users understand that they just use the http://... URL for their coordinator .. and not some other URL or endpoint and protocol...

the coordinator in the Trino cluster:

.. code-block:: text

    ./trino --server http://trino.example.com:8080

If successful, you will get a prompt to run the CLI. Use the ``help`` command to
see a list of supported commands:

.. code-block:: text

    trino> help

    Supported commands:
    QUIT
    EXIT
    CLEAR
    EXPLAIN [ ( option [, ...] ) ] <query>
        options: FORMAT { TEXT | GRAPHVIZ | JSON }
                TYPE { LOGICAL | DISTRIBUTED | VALIDATE | IO }
    DESCRIBE <table>
    SHOW COLUMNS FROM <table>
    SHOW FUNCTIONS
    SHOW CATALOGS [LIKE <pattern>]
    SHOW SCHEMAS [FROM <catalog>] [LIKE <pattern>]
    SHOW TABLES [FROM <schema>] [LIKE <pattern>]
    USE [<catalog>.]<schema>

You can now run SQL statements. After processing, the CLI will show results and
statistics.

Use the ``clear`` command to clear the terminal. To stop and exit the CLI, run
``exit`` or ``quit``.

.. code-block:: text

  trino> SELECT count(*) FROM tpch.tiny.nation;

  _col0
  -------
      25
  (1 row)

  Query 20220324_213359_00007_w6hbk, FINISHED, 1 node
  Splits: 13 total, 13 done (100.00%)
  2.92 [25 rows, 0B] [8 rows/s, 0B/s]

> In a typical deployment, authentication is required. The CLI supports multiple

Author: Removed.

:ref:`authentication methods <cli-authentication>`. The most common
authentication method is :ref:`cli-username-password-auth`.

As part of starting the CLI, you can set the default catalog and schema. This
allows you to query tables directly without specifying catalog and schema.

.. code-block:: text

  ./trino --server http://trino.example.com:8080 --catalog tpch --schema tiny

  trino:tiny> SHOW TABLES;

    Table
  ----------
  customer
  lineitem
  nation
  orders
  part
  partsupp
  region
  supplier
  (8 rows)

You can also set the default catalog and schema with the :doc:`/sql/use`

> statement.

Author: I added this example. Let me know if it isn't necessary.

Reviewer: This is really useful. thanks. let's keep this example.

      processing statistics.
  * - ``--disable-compression``
    - Disables compression of query results.
  * - ``--editing-mode``
    - Sets key bindings in the CLI to be compatible with VI or
      EMACS editors. Defaults to ``EMACS``.
  * - ``--http-proxy``
    - Configures the URL of the HTTP proxy to connect to Trino.
  * - ``--network-logging``
    - Configures the level of detail provided for network logging of the CLI.
      Defaults to ``NONE``, other options are ``BASIC``, ``HEADERS``, or
      ``BODY``.
  * - ``--password``
    - Prompts for a password. You can set the ``TRINO_PASSWORD`` environment
      variable with the password value to avoid the prompt. For more
      information, see :ref:`cli-username-password-auth`.
  * - ``--schema``
    - Sets the default schema. You can change the default catalog and schema
      with :doc:`/sql/use`.
  * - ``--server``
    - The HTTP/HTTPS address and port of the Trino coordinator. The port must be
      set to the port the Trino coordinator is listening for connections on.
      Trino server location defaults to ``http://localhost:8080``.
  * - ``--session``

>     - Sets one or multiple :ref:`session properties

Reviewer: nit: `one or more` is what I've seen written most often.

Reviewer 2: ok

      <session-properties-definition>`. Property can be used multiple times with

>       the format ``key=value``.

Reviewer 2: good idea

  * - ``--socks-proxy``
    - Configures the URL of the SOCKS proxy to connect to Trino.
  * - ``--source``
    - Specifies the name of the application or source connecting to Trino.
      Defaults to ``trino-cli``. The value can be used as input for
      :doc:`/admin/resource-groups`.
  * - ``--timezone``

>     - Sets session time zone. Defaults to ``America/New_York``.

Reviewer: Would be useful to mention it needs to be named zones an not abbreviations (PST) or offsets (+5:30). I'm not sure what best wording would be.

Reviewer 2: Is there a link to a doc that has all the named and supported values? Like a wikipedia page? Then we can link to that .. or maybe there are other docs for time properties in Trino that have docs around this?

Reviewer: https://en.wikipedia.org/wiki/IANA_time_zone_database is the page that explains a bit (and includes links to actual tzdb which has the names).

Reviewer 2: Lets link to here https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

  * - ``--trace-token``
    - Sets a token value for correlating requests and responses during

>       debugging.

Reviewer: Let's remove this since it's not something anyone uses and we don't have any supporting documentation guiding anyone on how to actually make use of this.

Reviewer 2: ok

this case, no other options are necessary, since the JVM running the CLI
recognizes these certificates.

Use the options from the following table to further configure TLS and
certificate usage:

.. list-table::
  :widths: 40, 60
  :header-rows: 1

  * - Option
    - Description
  * - ``--insecure``
    - Skip certificate validation when connecting with TLS/HTTPS (should only be
      used for debugging).
  * - ``--keystore-path``
    - The location of the Java Keystore file that contains the certificate of
      the server to connect with TLS.
  * - ``--keystore-password``
    - The password for the keystore. This must match the password you specified
      when creating the keystore.
  * - ``--keystore-type``

>     - Keystore type.

Reviewer: Maybe link to https://docs.oracle.com/en/java/javase/11/docs/specs/security/standard-names.html#keystore-types for SOME of the allowed values. Note that the values may differ by whatever JDK (and security providers) user has installed.

Reviewer: Same for `truststore-type`.

Reviewer: @mosabua please suggest wording to indicate that the values will vary by whatever JDK people use, so consult the similar docs for their JDK.

Reviewer 2: I would prefer to delay this whole addition since this is quite a bit more complicated and should maybe be handled in more detail in our JKS and PEM file page. I also dont know what is truly supported (e.g. do pem files work like on the server).  We could just link to our TLS page and the certificate section but it is not helpful enough at the moment either. Can we handle this all in a follow up ticket?

Reviewer: Sure, please create one to track.

Reviewer: bump @mosabua, please track this via an issue so that we don't forget to follow-up.

      the server to connect with TLS.
  * - ``--keystore-password``
    - The password for the keystore. This must match the password you specified
      when creating the keystore.
  * - ``--keystore-type``
    - Keystore type.
  * - ``--truststore-password``
    - The password for the truststore. This must match the password you
      specified when creating the truststore.
  * - ``--truststore-path``
    - The location of the Java truststore file that will be used to secure TLS.
  * - ``--truststore-type``
    - Truststore type.

.. _cli-authentication:

Authentication
--------------

The Trino CLI supports many :doc:`/security/authentication-types` detailed in
the following sections:

.. _cli-username-password-auth:

Username and password authentication
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Username and password authentication is typically configured in a cluster with
the ``PASSWORD`` authentication type.

If your Trino server requires password authentication, use the ``--password``
option to have the CLI prompt for a password. If your cluster uses a different
username, you can override your username with the ``--user`` option. It defaults

> to your operating system username.

Author: Added to `--password` option in table.

Username and password authentication is typically configured in a cluster using
the ``PASSWORD`` :doc:`authentication type </security/authentication-types>`,
for example with :doc:`/security/ldap` or :doc:`/security/password-file`.

The following code example connects to the server, establishes your user name,
and prompts the CLI for your password:

 .. code-block:: text
.. code-block:: text

  ./trino --server https://trino.example.com --user=myusername --password

You can set the ``TRINO_PASSWORD`` environment variable with the password value

> to avoid the prompt.

Author: It was moved to to the table as part of the `--password` description.

    ...
    at java.base/java.lang.Thread.run(Thread.java:834)
    select count(*) from tpch.tiny.nations

> 

Author: Add an anchor?

    at java.base/java.lang.Thread.run(Thread.java:834)
    select count(*) from tpch.tiny.nations

.. _cli-options:

Options
-------

The following table lists all supported command line options for the Trino CLI
in alphabetical order:

.. list-table::
  :widths: 40, 60
  :header-rows: 1

  * - Option
    - Description
  * - ``--access-token``
    - Access token for :ref:`cli-jwt-auth`.
  * - ``--catalog``
    - Default catalog.
  * - ``--client-info``
    - Extra information about client making query.
  * - ``--client-request-timeout``
    - Client request timeout (default: 2m).
  * - ``--client-tags``
    - Client tags.
  * - ``--debug``
    - Enable debug information.
  * - ``--disable-compression``
    - Disable compression of query results.
  * - ``--execute``
    - Execute specified statements and exit.
  * - ``--external-authentication``
    - Enable external authentication for :ref:`cli-external-sso-auth`.
  * - ``--external-authentication-redirect-handler``

>     - External authentication redirect handlers: DESKTOP_OPEN, SYSTEM_OPEN,

Reviewer 2: Set the external redirect handler for external authentication. Valid options are .. . 

Also make it a link to the section and we might need to add more info there.

      PRINT, OPEN, ALL (default: ALL).

>   * - ``--extra-credential``

Reviewer 2: What is that used for? Is there a section on this?

    - Extra credentials (property can be used multiple times; format is
      key=value).
  * - ``-f`` ``--file``

>     - Execute statements from file and exit.

Reviewer 2: really? Or does this have to be combined with --execute .. we should make an example section for this .. check out the stuff in the book .... pretty sure I wrote that up there

  * - ``-h`` ``--help``

>     - Show this help message and exit.

Reviewer 2: Show the help message and exit.

  * - ``--http-proxy``

>     - HTTP proxy to use for server connections.

Reviewer 2: Configure the HTTP proxy server hostname to use to connect to the server.

  * - ``--ignore-errors``

>     - Continue processing in batch mode when an error occurs (default is to exit

Reviewer 2: should link to the same section about batch mode where we talk about --execute

Author: In the output format section, there's an example where `--execute` is used, but batch mode isn't mentioned in that section. Batch mode is mentioned in the description for the `--output-format` option, though.

      immediately).
  * - ``--insecure``

>     - Skip validation of HTTP server certificates (should only be used for

Reviewer 2: Skip certificate validation , when connecting with TLS/HTTPS.

      debugging).
  * - ``--keystore-path``

>     - The location of the Java Keystore file that is used to secure TLS. Connect

Reviewer 2: The location of the Java keytstore file that contains the certificate of the server to connect with TLS.

>       to a cluster that uses :ref:`cli-certificate-auth`.

Reviewer 2: this is wrong I think.

Author: In the certificate authentication section, `--keystore-path` and `--keystore-password`` are mentioned as one of two CLI options to use to connect to a cluster that uses certificate authentication. Am I misinterpreting this section?

  * - ``--keystore-password``
    - The password for the keystore. This must match the password you specified
      when creating the keystore. Connect to a cluster that uses

>       :ref:`cli-certificate-auth`.

Reviewer 2: I think this is wrong too

Author: See above comment

  * - ``--keystore-type``

>     - Keystore type

Reviewer 2: what are valid values

Author: I don't understand what you're saying here.

  * - ``--krb5-config-path``

>     - Kerberos configuration files.

Reviewer 2: All these need to link to the kerberos auth section

Author: Just so you know, I deleted the links to that general section, initially because I was waiting until we add the content from the other Kerberos doc. I'm not sure if the links will go to specific content.

  * - ``--krb5-credential-cache-path``
    - Kerberos credential cache path.
  * - ``--krb5-disable-remote-service-hostname-canonicalization``
    - Disable service hostname canonicalization using the DNS reverse lookup.
  * - ``--krb5-keytab-path``
    - The location of the keytab that can be used to authenticate the principal
      specified by ``--krb5-principal``.
  * - ``--krb5-principal``
    - The principal to use when authenticating to the coordinator.
  * - ``--krb5-remote-service-name``
    - Trino coordinator Kerberos service name.
  * - ``--krb5-service-principal-pattern``
    - Remote kerberos service principal pattern (default: ${SERVICE}@${HOST})
  * - ``--network-logging``
    - Network logging level [NONE, BASIC, HEADERS, BODY] (default: NONE).
  * - ``--output-format``

>     - :ref:`cli-output-format` for batch mode [ALIGNED, VERTICAL, TSV,

Reviewer 2: We really need to create that batch mode section

Author: Yes. I've made a note to do that.

      TSV_HEADER, CSV, CSV_HEADER, CSV_UNQUOTED, CSV_HEADER_UNQUOTED, JSON,
      NULL] (default: CSV).
  * - ``--password``

>     - Prompt for a password, if your Trino server requires password

Reviewer 2: remove the "if..."

      authentication. For more information, see
      :ref:`cli-username-password-auth`.
  * - ``--progress``

>     - Show query progress in batch mode.

Reviewer 2: again .. need to incorporate that info into batch mode section

  * - ``--resource-estimate``

>     - Resource estimate (property can be used multiple times; format is

Reviewer 2: what is this?

Author: For displaying resource details? I really have no idea.

      key=value).
  * - ``--schema``
    - Default schema.
  * - ``--server``
    - The address and port of the Trino coordinator. The port must be set to
      the port the Trino coordinator is listening for HTTPS connections on.

>       Trino server location (default: localhost:8080). The Trino CLI does not

Reviewer 2: something is wrong here..

Author: This was copied from the CLI Kerberos authentication page. I'll simplify it.

>       support using ``http`` scheme for the URL when using LDAP authentication.

Reviewer 2: that has nothing to do with the CLI .. remove

  * - ``--session``
    - Session property (property can be used multiple times; format is

>       key=value; use 'SHOW SESSION' to see available properties).

Reviewer 2: link to session property docs and show session command

  * - ``--session-user``

>     - Username to impersonate.

Reviewer 2: Hm.. is there some section we need to write about impersonation?

Author: JWT authentication?

  * - ``--socks-proxy``
    - SOCKS proxy to use for server connections.
  * - ``--source``

>     - Name of source making query (default: trino-cli).

Reviewer 2: We should reword.. but not sure how .. lets chat and come up with something better

  * - ``--timezone``
    - Session time zone (default: America/New_York).
  * - ``--trace-token``

>     - Trace token.

Reviewer 2: hm.. whats that for?

  * - ``--truststore-password``

>     - The password for the truststore. This must match the password you

Reviewer 2: All these should be mentioned in tls section (and link to it here as well probably)

    - Resource estimate (property can be used multiple times; format is
      key=value)
  * - ``--schema``
    - Default schema
  * - ``--server``
    - The address and port of the Trino coordinator. The port must be set to
      the port the Trino coordinator is listening for HTTPS connections on.
      Trino server location (default: localhost:8080). The Trino CLI does not
      support using ``http`` scheme for the URL when using LDAP authentication
  * - ``--session``
    - Session property (property can be used multiple times; format is
      key=value; use 'SHOW SESSION' to see available properties)
  * - ``--session-user``
    - Username to impersonate
  * - ``--socks-proxy``
    - SOCKS proxy to use for server connections
  * - ``--source``
    - Name of source making query (default: trino-cli)
  * - ``--timezone``
    - Session time zone (default: America/New_York)
  * - ``--trace-token``
    - Trace token
  * - ``--truststore-password``
    - The password for the truststore. This must match the password you
      specified when creating the truststore.
  * - ``--truststore-path``
    - The location of the Java truststore file that will be used to secure TLS.

>   * - ``--truststore-type``

Author: Should the `--truststore` related options be mentioned in the TLS/HTTPS section at the top of the doc?

>   * - ``--truststore-type``

Author: @mosabua Should the --truststore related options be mentioned in the TLS/HTTPS section at the top of the doc?

Reviewer 2: You can do a list of those keystore and truststore properties ..

Reviewer 2: Something like ..

Further available options to configure TLS and certificate usage are ....

To access a Trino cluster configured to use :doc:`/security/jwt`, use the
``--access-token=<token>`` option to pass a JWT to the server.

.. _cli-kerberos-auth:

Kerberos authentication
^^^^^^^^^^^^^^^^^^^^^^^

In addition to the options that are required when connecting to an unauthorized
Trino coordinator, invoking the CLI with Kerberos support enabled requires a
number of additional command line options. The simplest way to invoke the CLI is
with a wrapper script.

.. code-block:: text

    #!/bin/bash

    ./trino \
      --server https://trino-coordinator.example.com:7778 \
      --krb5-config-path /etc/krb5.conf \
      --krb5-principal someuser@EXAMPLE.COM \
      --krb5-keytab-path /home/someuser/someuser.keytab \
      --krb5-remote-service-name trino \
      --keystore-path /tmp/trino.jks \
      --keystore-password password \
      --catalog <catalog> \
      --schema <schema>

The following table list the available options for Kerberos authentication:

.. list-table:: CLI options for Kerberos authentication
  :widths: 40, 60
  :header-rows: 1

  * - Option
    - Description
  * - ``--krb5-config-path``

>     - Kerberos configuration files.

Reviewer: ```suggestion
    - Path to Kerberos configuration files.
```

  * - ``--krb5-credential-cache-path``
    - Kerberos credential cache path.
  * - ``--krb5-disable-remote-service-hostname-canonicalization``
    - Disable service hostname canonicalization using the DNS reverse lookup.
  * - ``--krb5-keytab-path``
    - The location of the keytab that can be used to authenticate the principal
      specified by ``--krb5-principal``.

>   * - ``--krb5-principal``

Reviewer: While this list is ordered alphabetically maybe it'd be useful to order them by commonly used configs to more advanced ones? I don't have strong opinion since the list is short anyway.

Reviewer 2: I think we should leave as is

To access a Trino cluster configured to use :doc:`/security/jwt`, use the
``--access-token=<token>`` option to pass a JWT to the server.

.. _cli-kerberos-auth:

Kerberos authentication
^^^^^^^^^^^^^^^^^^^^^^^

In addition to the options that are required when connecting to a Trino
coordinator, that does not require Kerberos authentication, invoking the CLI
with Kerberos support enabled requires a number of additional command line

> options. The simplest way to invoke the CLI is with a wrapper script.

Reviewer 2: We want to get rid of the whole kerberos CLI doc and add the info from there in here .. but that is a much larger scope and we want to incrementally improve the docs .. we plan to do a follow up PR around this work. Same for LDAP CLI btw..

      --krb5-remote-service-name trino \
      --keystore-path /tmp/trino.jks \
      --keystore-password password \
      --catalog <catalog> \
      --schema <schema>

The following table list the available options for Kerberos authentication:

.. list-table:: CLI options for Kerberos authentication
  :widths: 40, 60
  :header-rows: 1

  * - Option
    - Description
  * - ``--krb5-config-path``
    - Kerberos configuration files.
  * - ``--krb5-credential-cache-path``
    - Kerberos credential cache path.
  * - ``--krb5-disable-remote-service-hostname-canonicalization``
    - Disable service hostname canonicalization using the DNS reverse lookup.
  * - ``--krb5-keytab-path``
    - The location of the keytab that can be used to authenticate the principal
      specified by ``--krb5-principal``.
  * - ``--krb5-principal``
    - The principal to use when authenticating to the coordinator.
  * - ``--krb5-remote-service-name``
    - Trino coordinator Kerberos service name.
  * - ``--krb5-service-principal-pattern``
    - Remote kerberos service principal pattern (default: ${SERVICE}@${HOST})

See :doc:`/security/cli` for more information on configuring and using Kerberos

> with the CLI.

Reviewer 2: Still true.. that doc still exists and has more info .. we want to move all that here .. but in follow up PR

> with the CLI.

Reviewer: Ah, I didn't notice that this is `installation/cli`. Thanks.
